File: Demos/Visual Basic .Net/09. Addendas/Adenda/Prolec.vb
Imports HyperSoft.Ejemplo.Data
Imports HyperSoft.ElectronicDocumentLibrary.Prolec.Addenda

Friend NotInheritable Partial Class Adenda
	
	Friend Shared Function Prolec(ByRef fileName As String) As Boolean
		'En este método se cargan los datos de la factura.
		Cfdi40.CargarDatosCompleto(electronicDocument)

		Dim addenda As Prolec = HyperSoft.ElectronicDocumentLibrary.Prolec.Addenda.Prolec.NewEntity()

		addenda.Data.Aviso.Value = "Aviso"
		addenda.Data.Divisa.Value = "Divisa"
		addenda.Data.OrdenCompra.Value = "OrdenCompra"
		addenda.Data.Proveedor.Value = "Proveedor"

		electronicDocument.Data.Addendas.Add(addenda)

		Return Save("Addenda_Prolec.xml", fileName)
	End Function

End Class

File: Demos/CSharp/09. Addendas/Adenda/Prolec.cs
using HyperSoft.Ejemplo.Data;
using HyperSoft.ElectronicDocumentLibrary.Prolec.Addenda;

namespace HyperSoft.Ejemplo.Adendas
{
  internal static partial class Adenda
  {
    internal static bool Prolec(out string fileName)
    {
      //En este método se cargan los datos de la factura.
      Cfdi40.CargarDatosCompleto(electronicDocument);

      Prolec addenda = HyperSoft.ElectronicDocumentLibrary.Prolec.Addenda.Prolec.NewEntity();

      addenda.Data.Aviso.Value = "Aviso";
      addenda.Data.Divisa.Value = "Divisa";
      addenda.Data.OrdenCompra.Value = "OrdenCompra";
      addenda.Data.Proveedor.Value = "Proveedor";

      electronicDocument.Data.Addendas.Add(addenda);

      return Save("Addenda_Prolec.xml", out fileName);
    }
  }
}